### PR TITLE
Addon repositories

### PR DESCRIPTION
**Pull request by ludeeus**
Ensure that we only handle valid add-on repositories.

### CODE REVIEW

File: supervisor/store/__init__.py [hunk 1]
```diff
@@ -60,12 +67,22 @@ async def _add_repository(url):
...
                 _LOGGER.error("Can't load from repository %s", url)
                 return
-            self.repositories[url] = repository
 
             # don't add built-in repository to config
             if url not in BUILTIN_REPOSITORIES:
+                # Verify that it is a add-on repository
+                repository_file = Path(repository.git.path, "repository.json")
+                try:
+                    SCHEMA_REPOSITORY_CONFIG(read_json_file(repository_file))
```
Comment (MartinHjelmare) [inline supervisor/store/__init__.py:76]: This does I/O so shouldn't be done in the event loop. Schedule it on the executor thread pool.

File: supervisor/store/__init__.py [hunk 2]
```diff
@@ -60,12 +67,24 @@ async def _add_repository(url):
...
             if url not in BUILTIN_REPOSITORIES:
+                # Verify that it is a add-on repository
+                repository_file = Path(repository.git.path, "repository.json")
+                try:
+                    await self.sys_run_in_executor(
+                        SCHEMA_REPOSITORY_CONFIG, read_json_file(repository_file)
+                    )
+                except (JsonFileError, vol.Invalid) as err:
+                    _LOGGER.error("%s is not a valid add-on repository. %s", url, err)
+                    await self.sys_run_in_executor(repository.remove)
```
[the remark this answers is not in the document]
Comment (MartinHjelmare) [reply supervisor/store/__init__.py:81]: That's just a side note, though.

[the remark this answers is not in the document]
Comment (ludeeus) [reply supervisor/store/__init__.py:81]: I don't see why it could not be

**Review by MartinHjelmare — APPROVED** [review]

**Review by MartinHjelmare — APPROVED** [review]
Looks good!

**Pull request merged**